src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -230,7 +230,18 @@ def dependencies(
         if entry is None:
             raise self._coordinate_not_found(key, coord)
 
-        return (entry, tuple(entries[(i.group, i.artifact)] for i in entry.dependencies))
+        dependency_entries = []
+        for d in entry.dependencies:
+            dependency = (d.group, d.artifact)
+            if dependency not in entries:
+                if d.packaging == "pom":
+                    # https://github.com/pantsbuild/pants/issues/20162#issuecomment-1813374036
+                    continue
+                else:
+                    raise RuntimeError(f"missing entry: {dependency}")
```

Comment:
I wonder if we should `continue` in this branch too.

In particular, AIUI, the lockfiles that Pants is parsing here are the output of coursier... so if coursier decides not to include an entry, then potentially Pants should just trust it did the right thing. 

i.e.

```suggestion
                    continue
```

I _imagine_ we shouldn't necessarily force users to add `jvm_artifact`s things that are mentioned but coursier doesn't feel the need to include. However, I don't know for sure, because I don't understand the JVM ecosystem fully 😄 

Although, if we do that, I'd suggest the code become something like this, stylistically (to avoid indexing the `entries` dict twice):

```python
for d in entry.dependencies:
    try:
        entry = entries[(d.group, d.artifact)]
    except KeyError:
        # <comment here>
        continue
    dependency_entries.append(entry)
```

Reply:
This makes sense, although I'm not a JVM expert either :)

Reply:
Done, ready for review